src/passes/Print.cpp

```diff
@@ -301,7 +301,7 @@ struct PrintSExpression : public UnifiedExpressionVisitor<PrintSExpression> {
   void visitLoop(Loop* curr);
   void visitTry(Try* curr);
   void visitTryTable(TryTable* curr);
-  void visitResume(Resume* curr);
```

Comment:
For the expressions that now depend on the types of their children to recover the printed type annotations, we now need to handle the case where the children are unreachable or have bottom heap types. See what we do for `StructNew` and friends just below this. It would also be good to add tests that exercise these cases.

Reply:
Please take a look at commits https://github.com/WebAssembly/binaryen/pull/7041/commits/157552ad4bcd77d7718fdfa944992b3505c416c4 and https://github.com/WebAssembly/binaryen/pull/7041/commits/480843d3b572e04b9b528b232af7dfc10971510e.